src/main/java/com/xpandit/plugins/xrayjenkins/services/enviromentvariables/XrayEnvironmentVariableSetter.java

```diff
@@ -0,0 +1,183 @@
+        final String testExecKeys = getModifiedTestExecutionsKeys(results, hostingType, logger);
+        final String testKeys = getModifiedTestKeys(results, hostingType, logger);
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_TEST_EXECS, testExecKeys);
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_TESTS, testKeys);
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_ISSUES_MODIFIED, getAllKeys(testExecKeys, testKeys));
+
+        return variableSetter;
+    }
+
+    /**
+     * Quick method that will return a Success XrayEnvironmentVariableSetter with no raw response.
+     *
+     * @return Success XrayEnvironmentVariableSetter with no raw response.
+     */
+    public static XrayEnvironmentVariableSetter success() {
+        return success(null);
+    }
+
+    /**
+     * Quick method that will return a Success XrayEnvironmentVariableSetter with a raw response.
+     *
+     * @return Success XrayEnvironmentVariableSetter with a raw response.
+     */
+    public static XrayEnvironmentVariableSetter success(@Nullable final String message) {
+        final XrayEnvironmentVariableSetter variableSetter = new XrayEnvironmentVariableSetter();
+
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_IS_REQUEST_SUCCESSFUL, TRUE_STRING);
+        if (message != null) {
+            variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_RAW_RESPONSE, message);
+        }
+
+        return variableSetter;
+    }
+
+    /**
+     * Quick method that will return a Failed XrayEnvironmentVariableSetter with no raw response.
+     *
+     * @return Failed XrayEnvironmentVariableSetter with no raw response.
+     */
+    public static XrayEnvironmentVariableSetter failed() {
+        return failed(null);
+    }
+
+    /**
+     * Quick method that will return a Failed XrayEnvironmentVariableSetter with a raw response.
+     *
+     * @return Failed XrayEnvironmentVariableSetter with a raw response.
+     */
+    public static XrayEnvironmentVariableSetter failed(@Nullable final String message) {
+        final XrayEnvironmentVariableSetter variableSetter = new XrayEnvironmentVariableSetter();
+
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_IS_REQUEST_SUCCESSFUL, FALSE_STRING);
+        if (message != null) {
+            variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_RAW_RESPONSE, message);
+        }
+
+        return variableSetter;
+    }
+
+    public void setAction(Run<?,?> build, TaskListener taskListener) {
+        setAction(build, taskListener.getLogger());
+    }
+
+    public void setAction(Run<?,?> build, @Nullable PrintStream logger) {
+        if (build != null) {
+            // Builds the same name, but with the name of each XrayEnvironmentVariable.
+            final Map<String, String> newVariablesByName = new HashMap<>();
+            for (Map.Entry<XrayEnvironmentVariable, String> entry : newVariables.entrySet()) {
```

Comment:
It is fine as it is, but can this code be refactored to a method called createVariablesNameCopy or similar? Up to you.